check_api/src/main/java/com/google/errorprone/util/SourceVersion.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2023 The Error Prone Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.errorprone.util;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import com.sun.tools.javac.code.Source;
+import com.sun.tools.javac.code.Source.Feature;
+import com.sun.tools.javac.util.Context;
+import java.util.Arrays;
+
+/**
+ * JDK source version utilities.
+ *
+ * @see RuntimeVersion
+ */
+public final class SourceVersion {
```

Comment:
Could also name the class something like `SourceFeature(s)`, though that might perhaps also call for slightly different method names.  :thinking: